boost-maven/boost-maven-plugin/src/main/java/io/openliberty/boost/docker/Dockerize.java

```diff
@@ -137,4 +141,55 @@ private void writeSpringBootDockerFile(File dockerFile, String libertyImage) thr
                 + "/config/dropins/spring/thin-${APP_FILE}");
         Files.write(dockerFile.toPath(), lines, Charset.forName("UTF-8"));
     }
+
+    public void createDockerIgnore() throws Exception {
+        if (springBootVersion != null) {
+            createSpringBootDockerIgnore();
+        } else {
+            throw new MojoExecutionException("Unable to create a .dockerignore file because application type is not supported");
+        }      
+    }
+
+    private void createSpringBootDockerIgnore() throws IOException {
+        File dockerIgnore = new File(projectDirectory, ".dockerignore");
+        boolean dockerIgnoreAlreadyExists = false;
+        try {
+            Files.createFile(dockerIgnore.toPath());
+            log.info("Creating .dockerignore: "+ dockerIgnore.getAbsolutePath());          
+        } catch (FileAlreadyExistsException e) {
+            log.warn("The .dockerignore file already exists");
+            dockerIgnoreAlreadyExists = true;
+        }
+        writeDockerIgnore(dockerIgnore, dockerIgnoreAlreadyExists);
+    }
+
+    private void writeDockerIgnore(File dockerIgnore, boolean dockerIgnoreAlreadyExists) throws IOException {
+        ArrayList<String> lines = new ArrayList<>();
+        if (pluginAlreadyExecuted(dockerIgnore)) {
+            return;
+        }
+
+        //Ignore the build.log getting generated by integration tests and the liberty runtime folder
+        if (dockerIgnoreAlreadyExists && dockerIgnore.length() > 0) {
+            lines.add("\n");
+        }
+        lines.add("# The following lines are added by boost-maven-plugin");           
+        lines.add("build.log");          
+        lines.add("target/liberty"); 
+        Files.write(dockerIgnore.toPath(), lines, Charset.forName("UTF-8"), StandardOpenOption.APPEND);
+    }
+
+    private boolean pluginAlreadyExecuted(File dockerIgnore) throws IOException {
+        try (FileReader fileReader = new FileReader(dockerIgnore)) {
+            try (BufferedReader bufferedReader = new BufferedReader(fileReader)) {
+                String line;
+                while ((line = bufferedReader.readLine()) != null) {
+                   if ("# The following lines are added by boost-maven-plugin".equals(line)) {
+                       return true;
+                   }
+                }
+            }
+        }
+        return false;
+    }
 }
```

Comment:
Two final, more general comments:

1. We should doc the behavior in the presence of existing .dockerignore(s) and also Dockerfile(s), at least in the form of a simple Javadoc comment in the mojo, which incorporates this into the help printed via  `mvn boost:help -Ddetail=true -Dgoal=docker-build`.      Just a simple one-line per behavior I think is a good start, all we need to do w/ this PR.

2. What to do about Gradle?  Assuming this doesn't get merged before the big Gradle code drop into this repo, we could just wait and keep going on this PR.   We could also open a separate issue.   Let's just do one or the other before closing this PR/issue.